nexus/reconfigurator/planning/src/example.rs

```diff
@@ -53,6 +56,10 @@ pub struct ExampleSystemBuilder {
     test_name: String,
     nsleds: usize,
     ndisks_per_sled: u8,
+    // None means nsleds
```

Comment:
I know this is a builder, but I think we'd be more likely to maintain the completeness of the `ExampleSystem` if we took a `Policy` as input, rather than duplicating what gets built by hand. I think it will also make the use of the `ExampleSystem` more consistent with production by driving things from `Policy` rather than another similar, but not quite the same interface.

We could then copy the policy into the `PlanningInput` when we build the `ExampleSystem`.

Reply:
That makes sense!

Reply:
Hmm, I agree in principle but it's a bit hard to express the current default of 1 zone per sled in a `Policy` struct. We could change the default after auditing everything, though.

How about I file a followup issue to do this? We could in principle do something like `with_policy(|policy| { policy.target_nexus_zone_count = 1; })` or whatever other knobs we want.

Reply:
> Hmm, I agree in principle but it's a bit hard to express the current default of 1 Nexus zone per sled in a `Policy` struct. We could change the default after auditing that our tests all set up the right number of Nexus zones, though.

I guess I assumed if there was enough sleds the planner would guarantee this, or that it would be the default.

> How about I file a followup issue to do this? I think once the 1-zone-per-sled issue is resolved, we can make the builder interface and `Policy` work together. We could in principle do something like `with_policy(|policy| { policy.target_nexus_zone_count = 1; })`.

A follow up seems fine to me.

Reply:
I'll file an issue, thanks!

Reply:
Filed #6803.